File: 392. Is Subsequence/isSubsequence.cpp
#include <iostream>

using namespace std;

class Solution {
    public:
        bool isSubsequence(string s, string t) {
            int p1 = 0; //for S
            int p2 = 0; //for T
    
            while(p1 < s.size()){
                if(p2 == t.size() && p1<s.size()) return false;
    
                if(s[p1] == t[p2]){
                    p1++;
                    p2++;
                }
                else{
                    p2++;
                }
            }
            return true;
        }
    };

File: 392. Is Subsequence/isSubsequence.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function (s, t) {
    let s_pointer = 0
    let t_pointer = 0
    while (s_pointer < s.length) {
        if (t_pointer === t.length && s_pointer < s.length) return false;

        if (s[s_pointer] === t[t_pointer]) {
            s_pointer++;
            t_pointer++;
        }
        else {
            t_pointer++;
        }
    }

    return true;
};